src/main/java/cart/service/CartService.java

```diff
@@ -0,0 +1,85 @@
+package cart.service;
+
+import cart.dto.ProductDto;
+import cart.entity.Cart;
+import cart.entity.Member;
+import cart.entity.Product;
+import cart.exception.customExceptions.DataNotFoundException;
+import cart.repository.dao.cartDao.CartDao;
+import cart.repository.dao.memberDao.MemberDao;
+import cart.repository.dao.productDao.ProductDao;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Transactional
+@Service
+public class CartService {
+
+    private final CartDao cartDao;
+    private final MemberDao memberDao;
+    private final ProductDao productDao;
+
+    public CartService(final CartDao cartDao, final MemberDao memberDao, final ProductDao productDao) {
+        this.cartDao = cartDao;
+        this.memberDao = memberDao;
+        this.productDao = productDao;
+    }
+
+    @Transactional(readOnly = true)
+    public List<ProductDto> findAllCartProductByEmail(final String email) {
+        final Optional<Member> findMember = memberDao.findByEmail(email);
+        if (findMember.isEmpty()) {
+            throw new DataNotFoundException("해당 사용자가 존재하지 않습니다.");
+        }
+        final Member member = findMember.get();
```

Comment:
```suggestion
        final Member member = memberDao.findByEmail(email)
          .orElseThrow(() -> new DataNotFoundException("해당 사용자가 존재하지 않습니다."));
```
Optional을 사용하니 이렇게 수정할 수 있을 것 같아요.
나머지도 마찬가지입니다. Optional에 대해서도 한번 공부해보면 좋겠네요.

Reply:
감사합니다.

옵셔널의 구조를 분석해봤는데요. 

Optional.EMPTY()를 쓰면 Optional의 기본생성자에 의해 생성된 value가 null인 옵셔널 객체를 반환하더군요.

그리고 `orElseThrow`는 해당 옵셔널의 value가 null일 시, 파라미터로 받은 Supplier를 실행 시킵니다..!

코드가 가독성이 좋게 바뀔 수 있겠군요!! 감사합니다.

Reply:
> 카트 추가 api나 카트 삭제 api에서 product_id를 path로 받도록 설계하신 이유를 알려주실 수 있을까요?

질문의 의도를 계속 생각하다가 제가 잘못설계한것 같아 이제야 답변남깁니다..!

혹시 질문하신 의도가 cart_id가 아닌, product_id로 넘긴 이유가 궁금하시다는 건가요!!?

처음 설계했을 때는, 유저에 대한 정보(email)와 상품에대한 식별자(product_id)만 있으면 cart table의  record를 특정할 수 있을 것이라 생각했습니다.
그런데, 지금 생각해보니 carts api는 carts/{리소스} 형태로 넘기는데 리소스가 product_id인게 이상한 것 같습니다.
cart_id가 되어야하는데 말이죠... 지금에서야 깨달았습니다.

지금 형태의 단점을 더 고민해봤을 때, 지금 설계는 email과 product_id를 받아
1. email로 -> member_id를 찾는다 (select)
2. member_id와 product_id로 해당 장바구니를 찾아 삭제한다(delete)
이렇게 2번 쿼리가 나갑니다.

만약 cart_id로 URI로 리소스를 넘겨줄 시, 
1. cart_id로 장바구니를 찾아 삭제한다(delete)
이렇게 1번 쿼리만 나갑니다.

종합해보면, URI에 장바구니에 대한 리소스를 표시했어야 하는데, 상품에 대한 리소스를 표시해서 api설계가 무너진(?) 느낌이다. 
또한, 쿼리문이 2번 나가서 효율이 떨어진다. 라는 결론에 이르렀습니다.

CartDto를 설계하여 한 번 리팩토링 해보겠습니다. 감사합니다.

Reply:
맞습니다. 말씀해주신 부분을 의도하고 코멘트를 남겨드리긴 했는데, email은 그대로 필요하지 않을까요?
요청 준 member의 cart인지를 확인하고 삭제해야할 것 같아서요.
resolver를 통해 email은 그대로 받도록 수정하시면 좋겠습니다.